redox/processor_neworder.go

```diff
@@ -242,6 +268,143 @@ func (o *newOrderProcessor) handleCreateAccount(ctx context.Context, order model
+	if tagNames == nil {
+		return nil, nil
+	}
+
+	existingTags := o.getExistingTags(match.Clinic)
+	for _, tagName := range tagNames {
+		_, exists := existingTags[tagName]
+		if !exists {
+			createTag := clinics.CreatePatientTagJSONRequestBody{
+				Name: tagName,
+			}
+			resp, err := o.clinics.CreatePatientTagWithResponse(ctx, *match.Clinic.Id, createTag)
+			if err != nil {
+				return nil, err
+			}
+			if resp.StatusCode() != http.StatusOK && resp.StatusCode() != http.StatusCreated {
+				return nil, fmt.Errorf("unexpected status code %v when creating tagName %s", resp.StatusCode(), tagName)
+			}
+		}
+	}
+
+	resp, err := o.clinics.GetClinicWithResponse(ctx, *match.Clinic.Id)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %vwhen fetching clinic with id %s", resp.StatusCode(), *match.Clinic.Id)
+	}
+
+	existingTags = o.getExistingTags(*resp.JSON200)
+	patientTagIds := clinics.PatientTagIds{}
+	for _, tagName := range tagNames {
+		patientTag, ok := existingTags[tagName]
+		if !ok {
+			// The tag should have been created. If it was deleted in the meantime returning an error will result in a retry
+			return nil, fmt.Errorf("patient tag doesn't exist")
+		}
+		patientTagIds = append(patientTagIds, *patientTag.Id)
+	}
+
+	return &patientTagIds, nil
+}
+
+func (o *newOrderProcessor) replacePatientTags(ctx context.Context, match clinics.EHRMatchResponse, tagIds *clinics.PatientTagIds) error {
+	if tagIds == nil {
+		return nil
+	}
+
+	patient := (*match.Patients)[0]
+	update := clinics.UpdatePatientJSONRequestBody{
+		Email:                          patient.Email,
+		BirthDate:                      patient.BirthDate,
+		FullName:                       patient.FullName,
+		Mrn:                            patient.Mrn,
+		TargetDevices:                  patient.TargetDevices,
+		Tags:                           tagIds,
+	}
+	resp, err := o.clinics.UpdatePatientWithResponse(ctx, *match.Clinic.Id, *patient.Id, update)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode() != http.StatusOK {
+		return fmt.Errorf("unexpected status code %v replacing tags for patient %s", resp.StatusCode(), *patient.Id)
+	}
+	return nil
+}
+
+func (o *newOrderProcessor) getPatientTagCodes(match clinics.EHRMatchResponse) map[string]struct{} {
+	result := make(map[string]struct{})
+	if match.Settings.Tags.Codes != nil {
+		for _, code := range *match.Settings.Tags.Codes {
+			result[code] = struct{}{}
+		}
+	}
+    return result
+}
+
+func (o *newOrderProcessor) getPatientTagsSeparator(match clinics.EHRMatchResponse) *string {
+	return match.Settings.Tags.Separator
+}
+
+func (o *newOrderProcessor) getTagNamesFromOrder(order models.NewOrder, match clinics.EHRMatchResponse) []string {
+	separator := o.getPatientTagsSeparator(match)
+	codes := o.getPatientTagCodes(match)
+	if len(codes) == 0 {
+		return nil
+	}
+
+	tagNames := make([]string, 0)
```

Comment:
Just a curiosity question if you know off of the top of your head:

Does this offer any benefit vs `tagNames := []string{}`? I find this a little clearer, and I think they're equivalent, but every time I think I've got the semantics of `make()` memorized I find small new edge case. :)

Reply:
The IntelliJ linter flags `[]string{}` for some reason. I don't know if there's any difference.